consensus/enclave/impl/src/lib.rs

```diff
@@ -401,7 +401,15 @@ impl SgxConsensusEnclave {
             }
 
             // The associated MintConfigTx should be valid.
-            self.validate_mint_config_txs(vec![mint_config_tx], current_block_index, config)?;
+            // Important note about the tombstone block: Since we are validating a
+            // MintConfigTx that has been previously added to the
+            // ledger, we shouldn't take the tombstone block into account. If we use
+            // current_block_index then a past MintConfigTx will eventually
+            // result in TombstoneBlockExceeded after enough blocks have been appended
+            // following it. The workaround here is to pretend that the current
+            // block index it is being validated against is within the tombstone range.
+            let config_block_index = mint_config_tx.prefix.tombstone_block.saturating_sub(1);
+            self.validate_mint_config_txs(vec![mint_config_tx], config_block_index, config)?;
```

Comment:
Looking closer it seems it's `mint_txs_with_config` versus `mint_config_txs`, but I'm unsure of the difference in behavior for the two types

Reply:
Then validating `MintConfigTxs` (`mint_config_txs`), that means we are validating *new* `MintConfigTxs` that are about to go into the ledger. As such, we need to ensure the tombstone block on them is not exceeded.

When validating `MintTxs` (`mint_txs_with_config`) we are validating the `MintTxs` against *old* `MintConfigTxs`, and the tombstone block of them might've been exceeded at this point - but we don't care about that since they are already in the ledger.